### ISSUE TITLE
querying users by email in login

### ISSUE THREAD

**Issue by cmsd2** [miltador/AspNetCore.Identity.DynamoDB#1]
In the sample project's account manager I'm trying this so I can lookup email instead of name:

```
var user = await _userManager.FindByEmailAsync(model.Email);
await _signInManager.PasswordSignInAsync(user, ...)
```
which does a _context.LoadAsync<TUser>(normalizedEmail)

I can't get that to work with the GSI -- it just returns null. Is it hitting the users table instead of the index?

so I tried changing it to use the Document model Query method:
```
Table table = Table.LoadTable(_client, Constants.DefaultTableName);
QueryOperationConfig config = new QueryOperationConfig() 
{
    IndexName = "Email.NormalizedValue-DeletedOn-index",
    KeyExpression = new Expression()
    {
        ExpressionStatement = "Email.NormalizedValue = :e",
        ExpressionAttributeValues = new Dictionary<string, DynamoDBEntry> {
            {":e", new Primitive(normalizedEmail)}
        }
    }
};
Search search = table.Query(config);
var results = await search.GetNextSetAsync(cancellationToken);
var userDocument = results.FirstOrDefault();
user = _context.FromDocument<TUser>(userDocument);
```

That rejects the KeyExpression because it is on a nested attribute of the Email Map attribute.

Any tips?

(testing this with the dynamodb docker container fwiw)

**Comment by miltador** [comment]
@cmsd2, good catch, man! 👍 
Tested it out and yes, seems like it is hitting table, not GSI. I'll dig into AWS SDK internals to see if it actually honors `IndexKey` setting on `LoadAsync`.

**Comment by miltador** [comment]
So, you are absolutely right. `LoadAsync` ignores `IndexName`. In fact, only `QueryAsync` and `ScanAsync` take into account passed `IndexName`.

> That rejects the KeyExpression because it is on a nested attribute of the Email Map attribute.

Exactly!  :(

http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SecondaryIndexes.html
> The key schema for the index. Every attribute in the index key schema must be a top-level attribute of type String, Number, or Binary. Other data types, including documents and sets, are not allowed. 

So at least `Email.NormalizedValue` should be top-level attribute. Then we can create secondary index of it and query on it.

**Comment by miltador** [comment]
@cmsd2 fixed by 6b48314

**Comment by cmsd2** [comment]
works great now, thanks!
